### PR TITLE
Ping using x/net/icmp

### PR DESCRIPTION
**Pull request by binjip978**

### CODE REVIEW

File: cmds/core/ping/ping.go [hunk 1]
```diff
@@ -44,145 +46,83 @@ var (
...
+	}
+	if echoReply.Seq != i {
+		return "", fmt.Errorf("got seq %v; want %v", echoReply.Seq, i)
+	}
+
+	return fmt.Sprintf("%d bytes from %v: icmp_seq=%v, time=%v", n, host, i, latency), nil
 }
 
 func main() {
+	_ = icmp.Echo{}
```
Comment (rminnich) [inline cmds/core/ping/ping.go:125]: This confused me, why was it needed? Is it to force inclusion of icmp? You can do that on the import line if so:
import _ "foo.bar"

**Comment by rminnich** [comment]
This is great. When I wrote the first version of this, I am not sure those packages existed ... this is a very nice improvement.

File: cmds/core/ping/ping.go (hunk 1 above)
Comment (binjip978) [reply cmds/core/ping/ping.go:125]: this is junk, I just want to see if I have free ride with testing it with raw socket under our CI, but I will probably need to provide testing interface :)

Comment (hugelgupf) [reply cmds/core/ping/ping.go:125]: A vmtest would be appropriate here. raw socket support will exist there

**Comment by binjip978** [comment]
Localhost will require more work, but current version has problem also.

**Comment by rminnich** [comment]
So is this ready or do you want to write the vmtest too?

**Comment by binjip978** [comment]
> So is this ready or do you want to write the vmtest too?

Yeah it's ready and  I added some vmtests, there is a known problem like pinging localhost with -c>1, but original version has the same problem. I have solution for IPv4, but not yet for IPv6.

**Review by rminnich — APPROVED** [review]

**Pull request merged**